src/urg_c_wrapper.cpp

```diff
@@ -31,83 +31,91 @@
  * Author: Chad Rockey, Mike O'Driscoll
  */
 
-#include "ros/ros.h"
-#include <urg_node/urg_c_wrapper.h>
+#include <urg_node/urg_c_wrapper.hpp>
+
 #include <limits>
+#include <memory>
 #include <string>
 #include <vector>
-#include <boost/crc.hpp>
+
+#include "boost/crc.hpp"
 
 namespace urg_node
 {
 
-URGCWrapper::URGCWrapper(const std::string& ip_address, const int ip_port,
-    bool& using_intensity, bool& using_multiecho, bool synchronize_time)
+URGCWrapper::URGCWrapper(
+  const EthernetConnection & connection, bool & using_intensity,
+  bool & using_multiecho, const rclcpp::Logger & logger)
+: ip_address_(connection.ip_address),
+  ip_port_(connection.ip_port),
+  serial_port_(""),
+  serial_baud_(0),
+  use_intensity_(using_intensity),
+  use_multiecho_(using_multiecho),
+  system_latency_(0),
+  user_latency_(0),
+  logger_(logger)
 {
-  // Store for comprehensive diagnostics
-  ip_address_ = ip_address;
-  ip_port_ = ip_port;
-  serial_port_ = "";
-  serial_baud_ = 0;
-
+  (void) adj_alpha_;
```

Comment:
This doesn't seem to be necessary since it is a class variable.  Same below.

Reply:
it somehow is. Otherwise my compiler complains about an unused class variable:
```
src/urg_node/include/urg_node/urg_c_wrapper.hpp:255:16: error: private field 'adj_alpha_' is not used [-Werror,-Wunused-private-field]
  const double adj_alpha_ = .01;
```

Now, I guess we could also just delete it all together, but as I am not familiar with what this was intended to do I opted for simply commenting it out.